### ISSUE TITLE
A problem when flipping a sprite

### ISSUE THREAD

**Issue by TroyBug** [photonstorm/phaser-ce#309]
Hi, @photonstorm  
I am a beginner of learning Phaser. I try to flip a sprite horizontally with this code:

        var dx = 70,
            dy = 30,
            game = new Phaser.Game(200,100,Phaser.CANVAS,'phaser-example',{
                  preload:preload,
                  create:create
            });
        function preload() {
            game.load.atlasJSONArray('c','1.png','1.json');
        }
        function create() {
            var c = game.add.sprite(dx,dy,'c');
            c.x = dx * 2;
            c.scale.x = -1;

            var anim = c.animations.add('attack');
            anim.play(10,true);
        }

It works:
![de1a90b8-0aa3-4155-9d8a-25dc45a20593](https://user-images.githubusercontent.com/9970058/28822706-dbea64f6-76ec-11e7-9baf-df5c7ef46c4c.GIF)
but I expected like this:
![ef](https://user-images.githubusercontent.com/9970058/28822747-100aa7aa-76ed-11e7-8bbd-432c9d1b94a8.PNG)
which their feet can keep the same position.

I am not sure if it is a bug, maybe I did something wrong?
How can I fix the position? Please, any help will be appreciated :)

**Comment by photonstorm** [comment]
You need to adjust the anchor of the sprite to be its center (`anchor.x = 0.5`), as position and scale are both based on this.

**Comment by TroyBug** [comment]
@photonstorm I modify the code

        function create() {
            var c = game.add.sprite(dx,dy,'c');
            c.smoothed = false;
            c.anchor.x = 0.5;
            c.scale.x = -1;

            var anim = c.animations.add('attack');
            anim.play(10,true);
        }
It is still not at the same position.
![d](https://user-images.githubusercontent.com/9970058/28826177-aafbb9b2-76fb-11e7-902a-27788f069050.png)
My sprite is 65X49 and packed by TexturePacker, it gave me a json file with some data like this:

        {
	        "filename": "00.png",
	        "frame": {"x":0,"y":0,"w":35,"h":38},
	        "rotated": false,
	        "trimmed": true,
	        "spriteSourceSize": {"x":3,"y":9,"w":35,"h":38},
	        "sourceSize": {"w":65,"h":49},
	        "pivot": {"x":0.5,"y":1}
        }

**Comment by photonstorm** [comment]
The feet will never be in the exact same place because they're not drawn in the center of the frame. Take a frame from the animation and flip it in Photoshop - are the feet in the middle? Or off to the side? because that is what scale.x does.

You can use anchor to adjust this offset, an anchor of 0.5 would cause the x position of the sprite to remain the same when flipped, but if the actual animation itself has been drawn to one edge of the frame then it cannot know about that and will need manually tweaking until it looks right.

tl;dr: scale.x works just like 'horizontal flip' in an art package. If it doesn't look right there, it won't look right in Phaser either.

**Comment by TroyBug** [comment]
@photonstorm Thank you very much for this detailed answer! I set anchor = 0.4 and it works good :)
